### ISSUE TITLE
Extra keys in state_dict and test failed.

### ISSUE THREAD

**Issue by OfficialBishal** [ArcInstitute/evo2#36]
Hi,

I ran the Evo2_40b model on test_evo2.py and test_evo2_generation.py. While executing, it outputs "Extra keys in state_dict".  Additionally, The Test Failed with high loss and low mean accuracy. 

**Is the extra keys in state_dict expected?**

```
Loading checkpoint shards for evo2_40b.pt
Found evo2_40b.pt
Extra keys in state_dict: {'blocks.20.mlp.l3._extra_state', 'blocks.28.mlp.l3._extra_state', 'blocks.22.out_filter_dense._extra_state', 'blocks.15.out_filter_dense._extra_state', 'blocks.41.mlp.l1._extra_state', 'blocks.20.out_filter_dense._extra_state', 'blocks.24.mlp.l2._extra_state', 'blocks.35.mlp.l1._extra_state', 'blocks.40.mlp.l2._extra_state', 'blocks.41.mlp.l3._extra_state', 'blocks.5.mlp.l2._extra_state', 'blocks.8.mlp.l1._extra_state', 'blocks.9.mlp.l2._extra_state', 'blocks.49.mlp.l1._extra_state', 'blocks.34.out_filter_dense._extra_state', 'blocks.2.out_filter_dense._extra_state', 'blocks.10.inner_mha_cls.Wqkv._extra_state', 'blocks.46.mlp.l3._extra_state', 'blocks.34.mlp.l1._extra_state', 'blocks.4.mlp.l3._extra_state', 'blocks.32.out_filter_dense._extra_state', 'blocks.35.mixer.attn._extra_state', 'blocks.27.out_filter_dense._extra_state', 'blocks.16.mlp.l3._extra_state', 'blocks.21.mlp.l3._extra_state', 'blocks.8.out_filter_dense._extra_state', 'blocks.12.mlp.l1._extra_state', 'blocks.15.mlp.l1._extra_state', 'blocks.11.mlp.l3._extra_state', 'blocks.18.mlp.l1._extra_state', 'blocks.31.mixer.dense._extra_state', 'blocks.25.out_filter_dense._extra_state', 'blocks.39.mlp.l2._extra_state', 'blocks.16.mlp.l2._extra_state', 'blocks.17.mixer.dense._extra_state', 'blocks.13.mlp.l3._extra_state', 'blocks.25.mlp.l2._extra_state', 'blocks.26.out_filter_dense._extra_state', 'blocks.31.mlp.l2._extra_state', 'blocks.49.mlp.l2._extra_state', 'blocks.35.mixer.dense._extra_state', 'blocks.17.mixer.attn._extra_state', 'blocks.18.mlp.l2._extra_state', 'blocks.28.mlp.l2._extra_state', 'blocks.7.out_filter_dense._extra_state', 'blocks.24.inner_mha_cls.Wqkv._extra_state', 'blocks.11.mlp.l1._extra_state', 'blocks.9.mlp.l3._extra_state', 'blocks.45.mlp.l1._extra_state', 'blocks.2.mlp.l2._extra_state', 'blocks.43.mlp.l3._extra_state', 'blocks.33.mlp.l3._extra_state', 'blocks.48.mlp.l3._extra_state', 'blocks.22.mlp.l3._extra_state', 'blocks.3.mlp.l1._extra_state', 'blocks.44.mlp.l1._extra_state', 'blocks.1.mlp.l3._extra_state', 'blocks.2.mlp.l1._extra_state', 'blocks.38.out_filter_dense._extra_state', 'blocks.12.out_filter_dense._extra_state', 'blocks.11.mlp.l2._extra_state', 'blocks.42.mlp.l1._extra_state', 'blocks.45.mlp.l3._extra_state', 'blocks.34.mlp.l3._extra_state', 'blocks.0.mlp.l1._extra_state', 'unembed.weight', 'blocks.22.mlp.l1._extra_state', 'blocks.19.out_filter_dense._extra_state', 'blocks.48.mlp.l1._extra_state', 'blocks.6.mlp.l2._extra_state', 'blocks.46.out_filter_dense._extra_state', 'blocks.49.inner_mha_cls.Wqkv._extra_state', 'blocks.32.mlp.l3._extra_state', 'blocks.13.mlp.l2._extra_state', 'blocks.49.mlp.l3._extra_state', 'blocks.26.mlp.l2._extra_state', 'blocks.27.mlp.l2._extra_state', 'blocks.37.mlp.l1._extra_state', 'blocks.6.mlp.l1._extra_state', 'blocks.13.out_filter_dense._extra_state', 'blocks.8.mlp.l2._extra_state', 'blocks.36.mlp.l3._extra_state', 'blocks.45.out_filter_dense._extra_state', 'blocks.12.mlp.l3._extra_state', 'blocks.20.mlp.l1._extra_state', 'blocks.16.out_filter_dense._extra_state', 'blocks.3.mlp.l2._extra_state', 'blocks.12.mlp.l2._extra_state', 'blocks.19.mlp.l1._extra_state', 'blocks.39.mlp.l1._extra_state', 'blocks.35.inner_mha_cls.Wqkv._extra_state', 'blocks.27.mlp.l1._extra_state', 'blocks.27.mlp.l3._extra_state', 'blocks.9.out_filter_dense._extra_state', 'blocks.6.out_filter_dense._extra_state', 'blocks.25.mlp.l1._extra_state', 'blocks.3.mixer.attn._extra_state', 'blocks.14.mlp.l3._extra_state', 'blocks.1.out_filter_dense._extra_state', 'blocks.9.mlp.l1._extra_state', 'blocks.15.mlp.l2._extra_state', 'blocks.19.mlp.l2._extra_state', 'blocks.34.mlp.l2._extra_state', 'blocks.43.out_filter_dense._extra_state', 'blocks.38.mlp.l2._extra_state', 'blocks.38.mlp.l1._extra_state', 'blocks.14.out_filter_dense._extra_state', 'blocks.42.mlp.l3._extra_state', 'blocks.31.mixer.attn._extra_state', 'blocks.24.mlp.l1._extra_state', 'blocks.26.mlp.l1._extra_state', 'blocks.17.mlp.l2._extra_state', 'blocks.49.mixer.dense._extra_state', 'blocks.29.mlp.l3._extra_state', 'blocks.35.mlp.l3._extra_state', 'blocks.39.mlp.l3._extra_state', 'blocks.13.mlp.l1._extra_state', 'blocks.33.mlp.l2._extra_state', 'blocks.7.mlp.l2._extra_state', 'blocks.17.mlp.l1._extra_state', 'blocks.29.mlp.l2._extra_state', 'blocks.23.mlp.l2._extra_state', 'blocks.29.out_filter_dense._extra_state', 'blocks.10.mlp.l2._extra_state', 'blocks.43.mlp.l2._extra_state', 'blocks.45.mlp.l2._extra_state', 'blocks.39.out_filter_dense._extra_state', 'blocks.25.mlp.l3._extra_state', 'blocks.40.mlp.l3._extra_state', 'blocks.47.out_filter_dense._extra_state', 'blocks.33.out_filter_dense._extra_state', 'blocks.44.mlp.l3._extra_state', 'blocks.0.mlp.l3._extra_state', 'blocks.18.mlp.l3._extra_state', 'blocks.46.mlp.l2._extra_state', 'blocks.18.out_filter_dense._extra_state', 'blocks.44.out_filter_dense._extra_state', 'blocks.3.mlp.l3._extra_state', 'blocks.17.mlp.l3._extra_state', 'blocks.37.mlp.l3._extra_state', 'blocks.32.mlp.l1._extra_state', 'blocks.20.mlp.l2._extra_state', 'blocks.41.mlp.l2._extra_state', 'blocks.31.mlp.l1._extra_state', 'blocks.46.mlp.l1._extra_state', 'blocks.36.mlp.l2._extra_state', 'blocks.15.mlp.l3._extra_state', 'blocks.38.mlp.l3._extra_state', 'blocks.24.mixer.attn._extra_state', 'blocks.30.mlp.l1._extra_state', 'blocks.48.out_filter_dense._extra_state', 'blocks.47.mlp.l2._extra_state', 'blocks.43.mlp.l1._extra_state', 'blocks.31.inner_mha_cls.Wqkv._extra_state', 'blocks.42.mixer.dense._extra_state', 'blocks.4.mlp.l2._extra_state', 'blocks.19.mlp.l3._extra_state', 'blocks.30.out_filter_dense._extra_state', 'blocks.24.mlp.l3._extra_state', 'blocks.28.out_filter_dense._extra_state', 'blocks.4.mlp.l1._extra_state', 'blocks.21.mlp.l1._extra_state', 'blocks.23.out_filter_dense._extra_state', 'blocks.3.mixer.dense._extra_state', 'blocks.48.mlp.l2._extra_state', 'blocks.1.mlp.l2._extra_state', 'blocks.24.mixer.dense._extra_state', 'blocks.1.mlp.l1._extra_state', 'blocks.22.mlp.l2._extra_state', 'blocks.42.inner_mha_cls.Wqkv._extra_state', 'blocks.11.out_filter_dense._extra_state', 'blocks.40.mlp.l1._extra_state', 'blocks.26.mlp.l3._extra_state', 'blocks.32.mlp.l2._extra_state', 'blocks.17.inner_mha_cls.Wqkv._extra_state', 'blocks.21.mlp.l2._extra_state', 'blocks.35.mlp.l2._extra_state', 'blocks.36.out_filter_dense._extra_state', 'blocks.5.out_filter_dense._extra_state', 'blocks.16.mlp.l1._extra_state', 'blocks.2.mlp.l3._extra_state', 'blocks.14.mlp.l1._extra_state', 'blocks.44.mlp.l2._extra_state', 'blocks.47.mlp.l3._extra_state', 'blocks.5.mlp.l1._extra_state', 'blocks.23.mlp.l3._extra_state', 'blocks.37.out_filter_dense._extra_state', 'blocks.37.mlp.l2._extra_state', 'blocks.0.mlp.l2._extra_state', 'blocks.42.mlp.l2._extra_state', 'blocks.49.mixer.attn._extra_state', 'blocks.10.mlp.l1._extra_state', 'blocks.7.mlp.l3._extra_state', 'blocks.8.mlp.l3._extra_state', 'blocks.3.inner_mha_cls.Wqkv._extra_state', 'blocks.42.mixer.attn._extra_state', 'blocks.41.out_filter_dense._extra_state', 'blocks.23.mlp.l1._extra_state', 'blocks.29.mlp.l1._extra_state', 'blocks.10.mixer.attn._extra_state', 'blocks.30.mlp.l2._extra_state', 'blocks.36.mlp.l1._extra_state', 'blocks.5.mlp.l3._extra_state', 'blocks.47.mlp.l1._extra_state', 'blocks.30.mlp.l3._extra_state', 'blocks.31.mlp.l3._extra_state', 'blocks.10.mixer.dense._extra_state', 'blocks.6.mlp.l3._extra_state', 'blocks.14.mlp.l2._extra_state', 'blocks.10.mlp.l3._extra_state', 'blocks.7.mlp.l1._extra_state', 'blocks.4.out_filter_dense._extra_state', 'blocks.0.out_filter_dense._extra_state', 'blocks.40.out_filter_dense._extra_state', 'blocks.33.mlp.l1._extra_state', 'blocks.21.out_filter_dense._extra_state', 'blocks.28.mlp.l1._extra_state'}

Sequence Results:
Sequence 1: Loss = 1.383, Accuracy = 25.03%
WARNING: Forward pass accuracy is below 50% on test sequence. Model may be broken, trained models should have >80% accuracy.
Sequence 2: Loss = 1.438, Accuracy = 23.88%
WARNING: Forward pass accuracy is below 50% on test sequence. Model may be broken, trained models should have >80% accuracy.
Sequence 3: Loss = 1.375, Accuracy = 28.51%
WARNING: Forward pass accuracy is below 50% on test sequence. Model may be broken, trained models should have >80% accuracy.
Sequence 4: Loss = 1.422, Accuracy = 24.92%
WARNING: Forward pass accuracy is below 50% on test sequence. Model may be broken, trained models should have >80% accuracy.

Mean Loss: 1.404
Mean Accuracy: 25.587%

Test Failed: Expected loss 0.216, got 1.404

```

**Comment by garykbrixi** [comment]
Hello Bishal, the extra states are expected.

However the failed test is concerning and suggests something is going wrong, perhaps in your environment or checkpoints.

Could you provide versions of transformer-engine and other packages you are using

See here for a docker with the pinned versions: https://github.com/Zymrael/vortex/pull/47/files
